src/domain/relayer/midnight/midnight_relayer.rs

```diff
@@ -0,0 +1,975 @@
+                TransactionRequest::new(transaction.id.clone(), transaction.relayer_id.clone()),
+                None,
+            )
+            .await?;
+
+        Ok(transaction)
+    }
+
+    async fn get_balance(&self) -> Result<BalanceResponse, RelayerError> {
+        // Get the ledger context from the sync service
+        let sync_service = self.sync_service.lock().await;
+        let context = sync_service.get_context();
+        drop(sync_service); // Drop the lock early
+
+        let wallet_seed = self.signer.wallet_seed();
+
+        let balance = self
+            .provider
+            .get_balance(wallet_seed, &context)
+            .await
+            .map_err(|e| RelayerError::ProviderError(format!("Failed to fetch balance: {}", e)))?;
+
+        let balance_u128 = balance
+            .to_string()
+            .parse::<u128>()
+            .map_err(|e| RelayerError::ProviderError(format!("Failed to parse balance: {}", e)))?;
+
+        Ok(BalanceResponse {
+            balance: balance_u128,
+            unit: MIDNIGHT_SMALLEST_UNIT_NAME.to_string(),
+        })
+    }
+
+    async fn get_status(&self) -> Result<RelayerStatus, RelayerError> {
+        let relayer_model = &self.relayer;
+
+        let nonce = self
+            .provider
+            .get_nonce(&relayer_model.address)
+            .await
+            .map_err(|e| {
+                RelayerError::ProviderError(format!("Failed to get account details: {}", e))
+            })?;
+
+        let nonce_str = nonce.to_string();
+
+        let balance_response = self.get_balance().await?;
+
+        let pending_statuses = [TransactionStatus::Pending, TransactionStatus::Submitted];
+        let pending_transactions = self
+            .transaction_repository
+            .find_by_status(&relayer_model.id, &pending_statuses[..])
+            .await
+            .map_err(RelayerError::from)?;
+        let pending_transactions_count = pending_transactions.len() as u64;
+
+        let confirmed_statuses = [TransactionStatus::Confirmed];
+        let confirmed_transactions = self
+            .transaction_repository
+            .find_by_status(&relayer_model.id, &confirmed_statuses[..])
+            .await
+            .map_err(RelayerError::from)?;
+
+        let last_confirmed_transaction_timestamp = confirmed_transactions
+            .iter()
+            .filter_map(|tx| tx.confirmed_at.as_ref())
+            .max()
+            .cloned();
+
+        Ok(RelayerStatus::Midnight {
+            balance: balance_response.balance.to_string(),
+            pending_transactions_count,
+            last_confirmed_transaction_timestamp,
+            system_disabled: relayer_model.system_disabled,
+            paused: relayer_model.paused,
+            nonce: nonce_str,
+        })
+    }
+
+    async fn delete_pending_transactions(
+        &self,
+    ) -> Result<DeletePendingTransactionsResponse, RelayerError> {
+        println!("Midnight delete_pending_transactions...");
```

Comment:
Let's use debug or info.

Println should be avoided in code due to logging configurations.